financial-connections/src/main/java/com/stripe/android/financialconnections/core/ComposeExtensions.kt

```diff
@@ -0,0 +1,65 @@
+package com.stripe.android.financialconnections.core
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.activity.ComponentActivity
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.stripe.android.financialconnections.di.FinancialConnectionsSheetNativeComponent
+import com.stripe.android.financialconnections.ui.FinancialConnectionsSheetNativeActivity
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlin.reflect.KProperty1
+
+/**
+ * Retrieves or builds a ViewModel instance, providing the [FinancialConnectionsSheetNativeComponent] (activity
+ * retained component) to the factory to facilitate its creation via dependency injection.
+ */
+@Composable
+internal inline fun <reified T : FinancialConnectionsViewModel<S>, S> paneViewModel(
+    factory: (FinancialConnectionsSheetNativeComponent) -> ViewModelProvider.Factory
+): T { return viewModel<T>(factory = factory(parentActivity().viewModel.activityRetainedComponent)) }
+
+/**
+ * Retrieves the parent [FinancialConnectionsSheetNativeActivity] from the current Compose context.
+ */
+@Composable
+internal fun parentActivity(): FinancialConnectionsSheetNativeActivity {
+    return extractActivityFromContext(LocalContext.current) as FinancialConnectionsSheetNativeActivity
+}
+
+private fun extractActivityFromContext(context: Context): ComponentActivity? {
```

Comment:
We have the same for the Payments SDK somewhere in `payments-core`. Should we move it to `stripe-core` and use it?